component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/dao/JWTStorageManager.java

```diff
@@ -126,17 +135,38 @@ public void persistJWTIdInDB(String jti, long expTime, long timeCreated) throws
         PreparedStatement preparedStatement = null;
         try {
             connection = IdentityDatabaseUtil.getDBConnection();
-            preparedStatement = connection.prepareStatement(Constants.SQLQueries.INSERT_JWD_ID);
+            if (JWTServiceDataHolder.getInstance().isPreventTokenReuse()){
+                preparedStatement = connection.prepareStatement(SQLQueries.INSERT_JWD_ID);
+            } else {
+                preparedStatement = connection.prepareStatement(SQLQueries.INSERT_OR_UPDATE_JWT_ID_MYSQL);
+                if (isH2DB()) {
+                    preparedStatement = connection.prepareStatement(SQLQueries.INSERT_OR_UPDATE_JWT_ID_H2);
+                } else if (isPostgreSQLDB()) {
+                    preparedStatement = connection.prepareStatement(SQLQueries.INSERT_OR_UPDATE_JWT_ID_POSTGRESQL);
+                } else if (isMSSqlDB() || isDB2DB()) {
+                    preparedStatement = connection.prepareStatement(SQLQueries.INSERT_OR_UPDATE_JWT_ID_MYSQL);
+                } else if (isOracleDB()) {
+                    preparedStatement = connection.prepareStatement(SQLQueries.INSERT_OR_UPDATE_JWT_ID_ORACLE);
+                }
+            }
             preparedStatement.setString(1, jti);
             Timestamp timestamp = new Timestamp(timeCreated);
             Timestamp expTimestamp = new Timestamp(expTime);
-            preparedStatement.setTimestamp(2, expTimestamp, Calendar.getInstance(TimeZone.getTimeZone(Constants.UTC)));
+            preparedStatement.setTimestamp(2, expTimestamp,
+                    Calendar.getInstance(TimeZone.getTimeZone(Constants.UTC)));
             preparedStatement.setTimestamp(3, timestamp,
                     Calendar.getInstance(TimeZone.getTimeZone(Constants.UTC)));
+            if (isOracleDB()) {
+                preparedStatement.setString(4, jti);
+                preparedStatement.setTimestamp(5, expTimestamp,
+                        Calendar.getInstance(TimeZone.getTimeZone(Constants.UTC)));
+                preparedStatement.setTimestamp(6, timestamp,
+                        Calendar.getInstance(TimeZone.getTimeZone(Constants.UTC)));
+            }
             preparedStatement.executeUpdate();
             preparedStatement.close();
             connection.commit();
-        } catch (SQLException e) {
+        } catch (SQLException|DataAccessException e) {
```

Comment:
Let's throw 'e' as well.

Reply:
Fixed in [88ef3d2](https://github.com/wso2-extensions/identity-oauth-addons/pull/98/commits/88ef3d2d1aa8339687402302ac2384d24888d8d9)